client/my-sites/plans/current-plan/current-plan-thank-you-card/paid-plan-thank-you-card.js

```diff
@@ -1,97 +1,206 @@
-									</h1>
-									<p>
-										{ translate(
-											'We’ve finished setting up spam filtering and backups for you.'
-										) }
-										<br />
-										{ translate( "You're now ready to finish the rest of the checklist." ) }
-									</p>
-									<Button primary href={ this.getMyPlanRoute() }>
-										{ translate( 'Continue' ) }
-									</Button>
-								</Fragment>
-							) : (
-								<Fragment>
-									<img
-										className="current-plan-thank-you-card__illustration"
-										alt=""
-										aria-hidden="true"
-										src="/calypso/images/illustrations/fireworks.svg"
-									/>
-									<h1 className="current-plan-thank-you-card__title">
-										{ translate( 'Thank you for your purchase!' ) }
-									</h1>
-									<p>{ translate( "Now let's make sure your site is protected." ) }</p>
-									<p>
-										{ translate(
-												"We're setting up spam filters and site backups for you first. Once that's done, our security checklist will guide you through the next steps."
-										) }
-									</p>
-
-									<ProgressBar isPulsing total={ 100 } value={ progressComplete || 0 } />
+				{ installState === INSTALL_STATE_UNCOMPLETE && (
+					<ThankYouCard
+						illustration={ fireworksIllustration }
+						showHideMessage
+						title={ translate( 'Thank you for your purchase!' ) }
+					>
+						<p>{ translate( "Now let's make sure your site is protected." ) }</p>
+						<p>
+							{ preventWidows(
+								translate(
+									"We're setting up spam filters and site backups for you first. Once that's done, our security checklist will guide you through the next steps."
+								)
+							) }
+						</p>
 
-									<p>
-										<a href={ this.getMyPlanRoute() }>{ translate( 'Hide message' ) }</a>
-									</p>
-								</Fragment>
+						<ProgressBar isPulsing total={ 100 } value={ installProgress || 0 } />
+					</ThankYouCard>
+				) }
+				{ installState === INSTALL_STATE_COMPLETE && (
+					<ThankYouCard
+						illustration={ securityIllustration }
+						showContinueButton
+						title={ translate( 'So long spam, hello backups!' ) }
+					>
+						<p>
+							{ preventWidows(
+								translate( 'We’ve finished setting up spam filtering and backups for you.' )
 							) }
-						</div>
-					</Card>
+							<br />
+							{ preventWidows(
+								translate( "You're now ready to finish the rest of the checklist." )
+							) }
+						</p>
+					</ThankYouCard>
 				) }
 			</Fragment>
 		);
 	}
 }
 
 export default connect( state => {
+	const site = getSelectedSite( state );
 	const siteId = getSelectedSiteId( state );
 
+	const installProgress = getJetpackProductInstallProgress( state, siteId );
+
+	let installState;
+	// @TODO we'll need a way to detect generic error states here and add `INSTALL_STATE_ERRORED`
+	if ( installProgress === 100 ) {
+		installState = INSTALL_STATE_COMPLETE;
+	} else {
+		installState = INSTALL_STATE_UNCOMPLETE;
```

Comment:
As is, these don't seem to add any value, but I understand they're here in anticipation of adding error handling. That will be in a follow-up PR?

Reply:
Correct, that's the plan. I removed some previous unfinished work on it here: https://github.com/Automattic/wp-calypso/pull/33228/commits/a5ec0320d964fd1d0bd460dc8204ae18444e33f1 and will follow up in another PR.